programs/graph_program/src/instructions/make_connection.rs

```diff
@@ -1,17 +1,22 @@
-use crate::{constants::CONNECTION_SEED, state::Connection};
+use crate::{state::{Connection, ConnectionStatus}, constants::*};
 
 use anchor_lang::prelude::*;
 
 #[derive(Accounts)]
 #[instruction(to: Pubkey)]
 pub struct MakeConnection<'info> {
     #[account(
-        init,
+        // It is safe to init if neeeded (overwrite existing account [or reinitialize])
+        // because we need this behavior so if anyone revoked the connection but
+        // they didn't close the account they can still reconnect.
+        init_if_needed,
         payer = from,
-        seeds = [CONNECTION_SEED, from.key().as_ref(), to.as_ref()],
+        space = Connection::calculate_space(),
+        seeds = [CONNECTION_SEED.as_ref(), from.key().as_ref(), to.as_ref()],
         bump,
     )]
     pub connection: Account<'info, Connection>,
+    pub clock: Sysvar<'info, Clock>,
```

Comment:
Read in the anchor docs its recommended to get access to the Clock this way. Available on the prelude of anchor_lang.

https://github.com/metaplex-foundation/metaplex-program-library/blob/master/auction-house/program/src/receipt/mod.rs#L131
https://docs.rs/anchor-lang/latest/anchor_lang/prelude/struct.Clock.html

Reply:
Alright! Yeah this is what I found on their repos but we have to pass in system program every time we want to use this one.

Reply:
Addressed